src/frontend/src/test-e2e/register.test.ts

```diff
@@ -231,6 +237,61 @@ test("Log into client application, after registration", async () => {
   });
 }, 300_000);
 
+test("Add key type on login if unknown", async () => {
+  await runInBrowser(async (browser: WebdriverIO.Browser) => {
+    const authenticatorId1 = await addVirtualAuthenticator(browser);
+
+    await browser.url(II_URL);
+    const userNumber = await FLOWS.registerNewIdentityWelcomeView(
+      DEVICE_NAME1,
+      browser
+    );
+
+    const credentials = await getWebAuthnCredentials(browser, authenticatorId1);
+    expect(credentials).toHaveLength(1);
+
+    // check device in II canister
+    const agent = new HttpAgent({
+      host: "http://127.0.0.1:4943",
+      // @ts-ignore
+      fetch,
+    });
+    await agent.fetchRootKey();
+    const actor = Actor.createActor<_SERVICE>(internet_identity_idl, {
+      agent,
+      canisterId: II_CANISTER_ID,
+    });
+    let devices = await actor.lookup(BigInt(userNumber));
+    expect(devices).toHaveLength(1);
+    expect(devices[0].key_type).toHaveProperty("unknown");
+
+    // sign in
+    const demoAppView = new DemoAppView(browser);
+    await demoAppView.open(TEST_APP_NICE_URL, II_URL);
+    await demoAppView.waitForDisplay();
+    await demoAppView.signin();
+
+    const authenticatorId2 = await switchToPopup(browser);
+    await addWebAuthnCredential(
+      browser,
+      authenticatorId2,
+      credentials[0],
+      originToRelyingPartyId(II_URL)
+    );
+    const authenticateView = new AuthenticateView(browser);
+    await authenticateView.waitForDisplay();
+    await authenticateView.pickAnchor(userNumber);
+
+    // wait for the device to be updated by the front-end
+    await new Promise((r) => setTimeout(r, 2000));
```

Comment:
that's reliable? the CI won't fail time to time?

Reply:
It seems so. If it turns out to start failing, we can still implement retries and / or a longer timeout.